force-app/main/default/classes/HOT_UpdateChildRecordsService.cls

```diff
@@ -125,108 +87,253 @@ public without sharing class HOT_UpdateChildRecordsService {
             ]
         );
 
-        //Updates WorkOrders
-        for (WorkOrder workOrder : workOrders.values()) {
-            if (input.isAccountChoice) {
+        if (input.request != null && input.workOrder == null) {
+            updateRequest(
+                request,
+                input.isAccountChoice,
+                input.isAddressChoice,
+                input.isSubjectChoice,
+                input.isDescriptionChoice,
+                input.isScreenInterpreterChoice,
+                input.isDispatcherChoice,
+                input.isInterpreterMethodChoice
+            );
+            updateWorkOrder(
+                workOrders,
+                request,
+                input.isAccountChoice,
+                input.isAddressChoice,
+                input.isSubjectChoice,
+                input.isDescriptionChoice,
+                input.isInterpreterMethodChoice
+            );
+            updateWorkOrderLineItem(
+                workOrderLineItems,
+                request,
+                input.isAddressChoice,
+                input.isSubjectChoice,
+                input.isDescriptionChoice,
+                input.isInterpreterMethodChoice
+            );
+            updateServiceAppointments(
+                serviceAppointments,
+                request,
+                input.isAccountChoice,
+                input.isAddressChoice,
+                input.isSubjectChoice,
+                input.isDescriptionChoice,
+                input.isScreenInterpreterChoice,
+                input.isDispatcherChoice,
+                input.isInterpreterMethodChoice
+            );
+        } else if (input.request != null && input.workOrder != null) {
+            list<WorkOrder> singleWorkOrder = new List<WorkOrder>{ input.workOrder };
+            updateWorkOrder(
+                singleWorkOrder,
+                request,
+                input.isAccountChoice,
+                input.isAddressChoice,
+                input.isSubjectChoice,
+                input.isDescriptionChoice,
+                input.isInterpreterMethodChoice
+            );
+            updateWorkOrderLineItem(
+                workOrderLineItems,
+                request,
+                input.isAddressChoice,
+                input.isSubjectChoice,
+                input.isDescriptionChoice,
+                input.isInterpreterMethodChoice
+            );
+            updateServiceAppointments(
+                serviceAppointments,
+                request,
+                input.isAccountChoice,
+                input.isAddressChoice,
+                input.isSubjectChoice,
+                input.isDescriptionChoice,
+                input.isScreenInterpreterChoice,
+                input.isDispatcherChoice,
+                input.isInterpreterMethodChoice
+            );
+        }
+        return errors;
```

Comment:
Ser du har en try/catch for hvert nivå (R, WO,...), hva om en eller flere av disse feiler. Hvordan sender du into tilbake til brukeren? Dumt om de tror alt funket, så ble f.eks. ikke SA'ene oppdatert.